### PR TITLE
Implement mapdb context manager using Jackson for serialization

### PR DESCRIPTION
**Pull request by veptechno**
Now MapDbBotContextManager uses the serialization mechanism provided by java.io. This is inconvenient to use because it forces you to implement the Serializable interface for all classes that are stored in BotContext.

This PR adds a BotContextManager that uses MapDb and Jackson for serialization. 
**What's done**: created JacksonMapDbBotContextManager class, deprecated MapDbBotContextManager class, added tests for JacksonMapDbBotContextManager.

### CODE REVIEW

**Review by Denire — APPROVED** [review]

File: managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/BotContextModel.kt [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 import com.justai.jaicf.context.BotContext
+import com.justai.jaicf.context.DialogContext
 import java.io.Serializable
 
-class BotContextModel(botContext: BotContext): Serializable {
-    val dialogContext = botContext.dialogContext
-    val result: Any? = botContext.result
-    val client = botContext.client.toMap()
-    val session = botContext.session.toMap()
+internal class BotContextModel private constructor(): Serializable {
```
Comment (nikvoloshin) [inline managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/BotContextModel.kt:7]: Did you check this change with the old MapDB manager? As far as old MapDB works with java.io.Serializable we should double-check all changes in serializable classes

Comment (nikvoloshin) [reply managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/BotContextModel.kt:7]: Also, we shouldn't change the visibility of public classes.
And anyway, I'd suggest remaining all old code unchanged and mark it deprecated.
And implement jackson-mapdb as completely new manager

File: managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/MapDbBotContextManager.kt [hunk 2]
```diff
@@ -8,7 +8,14 @@ import com.justai.jaicf.context.manager.BotContextManager
 import org.mapdb.DBMaker
 import org.mapdb.Serializer
 
-class MapDbBotContextManager(dbFilePath: String? = null): BotContextManager {
+@Deprecated(
+    "MapDbBotContextManager is deprecated. Use JacksonMapDbBotContextManager instead",
+    ReplaceWith(
```
Comment (nikvoloshin) [inline managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/MapDbBotContextManager.kt:13]: Let's not add replace with here. If a user will decide to use this quick fix, he will break his db-file

Comment (nikvoloshin) [reply managers/mapdb/src/main/kotlin/com/justai/jaicf/context/manager/mapdb/MapDbBotContextManager.kt:13]: Also, let's add incompatibility notice in deprecation message

**Review by nikvoloshin — CHANGES_REQUESTED** [review]

**Review by nikvoloshin — APPROVED** [review]

**Pull request merged**